src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -138,6 +138,49 @@ public VisitArgumentResult(VisitResult visitResult, Optional<LocalState> stateFo
             }
         }
 
+        private static class PooledDictionaryWithCustomBoundDefaultExpressionComparison
+        {
+            private sealed class CustomEqualityComparer : IEqualityComparer<object>
+            {
+                public static CustomEqualityComparer Instance { get; } = new();
+
+                private CustomEqualityComparer()
+                {
+                }
+
+                public new bool Equals([AllowNull] object x, [AllowNull] object y)
+                {
+                    if (x is BoundDefaultExpression boundDefaultExpression1 &&
```

Comment:
It seems like in some cases we will need a 'BoundDefaultExpression' to compare equal to a 'BoundLiteralExpression'. Also, depending on how the `BoundLiteral` is used, you might only be able to get its type by checking the conversion it is wrapped in. I don't completely recall.

Reply:
Yes, sorry :)